### PR TITLE
Include web2010 in OpenShift

### PR DESCRIPTION
**Pull request by obieler**
**Description**

Include web2010 in OpenShift (wwp-test/wwp projects).

See: 
* https://confluence.epfl.ch:8443/display/idevfsd/Web2010+-+Ramener+web2010+dans+OpenShift
* PR related: https://github.com/epfl-si/web2010/pull/92

**Notes**

* There is volontarily no OC Route, because traffic will be routabe trough Varnish.
  * Varnish config adaptation for web2010 are currently not deployed on wwp + wwp-test.
* I put anyway a temporary route (nginx-web2010) on wwp-test to test if needed.
  * Simply point locally www-test-web2010.epfl.ch to 128.178.222.83 (/etc/hosts)
    * Ex url: https://www-test-web2010.epfl.ch/css/epfl.css

**Deploy process** (can be deployed separately to WP):

```
`./ansible/wpsible --tags "web2010" --prod`  
`./ansible/wpsible --tags "web2010.rebuild"`  
`./ansible/wpsible --tags "web2010.image.promote" --prod`
```

### CODE REVIEW

**Comment by domq** [comment]
Reviewing now

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      selector:
+        deployment-config.name: nginx-web2010
+      template:
+        metadata:
+          creationTimestamp: null
+          labels:
+            deployment-config.name: nginx-web2010
+        spec:
+          containers:
+            - image: 'docker-registry.default.svc:5000/{{openshift_namespace}}/nginx-web2010:{{ web2010_image_tag }}'
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:63]: Please make this a variable

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+              imagePullPolicy: Always
+              name: nginx-web2010
+              resources: {}
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:66]: Please remove all the “contingent” stuff, and set the minimum number of lines that gets the pod to do what you want — Or in other words, use the default values to the maximum extent possible.

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    state: latest
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: nginx-web2010
+      namespace: "{{ openshift_namespace }}"
+    spec:
+      type: ClusterIP
+      ports:
+        - name: http
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:119]: Do you need two ports? If not, please remove one.

**Review by domq — CHANGES_REQUESTED** [review]
Excellent work!

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml (hunk 3 above)
Comment (obieler) [reply ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:117]: Ok, removed in https://github.com/epfl-si/wp-ops/pull/492/commits/9205300e8126fbe47a26fe07025b32cee9148d4c.

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml (hunk 2 above)
Comment (obieler) [reply ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml]: Fixed in https://github.com/epfl-si/wp-ops/pull/492/commits/9205300e8126fbe47a26fe07025b32cee9148d4c.

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml (hunk 1 above)
Comment (obieler) [reply ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml]: Done in https://github.com/epfl-si/wp-ops/pull/492/commits/9205300e8126fbe47a26fe07025b32cee9148d4c.

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        metadata:
+          creationTimestamp: null
+          labels:
+            deployment-config.name: nginx-web2010
+        spec:
+          containers:
+            - image: "{{ web2010_image_os_path }}"
+              imagePullPolicy: Always
+              name: nginx-web2010
+              terminationMessagePath: /dev/termination-log
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:66]: There's a bit of default-value leftovers here ☺ I don't think we need termination messages for a Web server.

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+              terminationMessagePolicy: File
+              readinessProbe:
```
Comment (domq) [inline ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:68]: Please use either [HTTP readiness / liveness probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request), or no probes at all.

**Review by domq — CHANGES_REQUESTED** [review]
Thanks! I'm afraid I found some more...

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml (hunk 5 above)
Comment (obieler) [reply ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml:66]: Updated in https://github.com/epfl-si/wp-ops/pull/492/commits/d7b130319eba67f312a646599fec73b40d5c576a.

File: ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml (hunk 4 above)
Comment (obieler) [reply ansible/roles/wordpress-openshift-namespace/tasks/web2010.yml]: Removed in https://github.com/epfl-si/wp-ops/pull/492/commits/442389e703a5c60ae4fb2a033e64b578b51d6b4f.

**Review by domq — APPROVED** [review]

**Pull request merged**